### DISCUSSION TITLE
I18n: Changing Language

### DISCUSSION THREAD

**Discussion by chris-der-kleine** [Pryaxis/TShock#1745] [Q&A] [answered]
Hey everbody,

I'm gonna be hosting an TShock Server and I want to customize all the messages to be in another language.

When looking through `Localization/EnglishLanguage.cs` there seems no easy way to at least change the language
The `-lang` Attribute doesn't seem to work either

Has anyone an idea how I can solve this problem?

**Comment by hakusaro** [comment, marked as answer]
i18n is, admittedly, pretty low on our list right now. We're aware of the problems with just using English, particularly for our users in China, and while we want to fix this, we don't really have concrete plans to make i18n easier. TShock in particular predates the localization system and many of the settings rely on English language being in use.

We're always open to pull requests to help us softcode strings.

**Comment by AxeelAnder** [comment]
I'll work on logging stuff along with i18n after we formally get into .netcore

**Reply by dredhorse** [reply]
what would it take to support for example german?
Just a GermanLanguage.cs and a change in TSock.cs to support a configurable language instead of hardcoding it?
